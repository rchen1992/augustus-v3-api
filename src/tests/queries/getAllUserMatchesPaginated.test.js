const mockMatchRepo = require('@tests/mockRepos/match');
const mockUserRepo = require('@tests/mockRepos/user');
const { testQuery } = require('@tests/utils');

testQuery({
    id: "Get all of a user's matches paginated with offset and limit",
    query: `
        {
            user(id: 2) {
                id
                matches(offset: 1, limit: 2) {
                    id
                    createdAt
                }
            }
        }
    `,
    variables: {},
    context: () => {
        return {
            repos: {
                match: mockMatchRepo,
                user: mockUserRepo,
            },
        };
    },
    expected: undefined,
});
