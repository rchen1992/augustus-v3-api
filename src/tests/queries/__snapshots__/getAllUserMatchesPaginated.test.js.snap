// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query: Get all of a user's matches paginated with offset and limit 1`] = `
Object {
  "data": Object {
    "user": Object {
      "id": "2",
      "matchCount": 4,
      "matches": Array [
        Object {
          "createdAt": "2018-11-03 05:46:42",
          "id": "7",
        },
        Object {
          "createdAt": "2018-09-17 01:38:13",
          "id": "4",
        },
      ],
    },
  },
}
`;
